generatorv2/README.md

```diff
@@ -0,0 +1,134 @@
+Since that can lead to errors in the generator.
+
+Bad way:
+
+```toml
+[1e22cceb-c5e4-4562-9afe-aef07ad1eaf4]
+description = "basic"
+include = false
+include = false
+```
+
+Good way:
+
+```toml
+[1e22cceb-c5e4-4562-9afe-aef07ad1eaf4]
+description = "basic"
+include = false
+```
+
+### Template
+
+The generator uses a template file to generate the test file.
+The template is located under the `.meta` for each exercise.
+
+This template has to be manually written for each exercise.
+The goal although is to make it so that you only have to write the template once and then it will be able to be used to generate new tests.
+
+The template file is written in [Embedded Ruby(ERB)][erb].
+ERB enables you to write Ruby code inside of the template file.
+It also means that the templates can be highly customizable since you can write any Ruby code you want.
+
+When writing the template file, it is recommended to look at already existing template files to get a better understanding of how it works.
+The template is getting a slightly modified version of the canonical data, so you can check out the [canonical data][canonical data] to see the data structure.
+The modification is that the cases which are not included in the toml file will be removed from the data structure.
+
+When writing the template so is it a special tool that can help with giving `# skip` and `skip` tags for tests.
+You simply have to call the `status` method.
+It will return either `# skip` or `skip` depending on if it is the first test case or not.
+
+Here is an example:
+
+```
+<%= status()%>
+<%= status()%>
+<%= status()%>
+```
+
+result:
+
+```
+# skip
+skip
+skip
+```
+
+### The Test Generator
+
+If all the earlier steps are done so can you run the generator.
+To run the generator you need to have a working Ruby installation and installed all gems in the Gemfile.
+The generator is located in the `bin` directory and is called `generator.rb`.
+
+To run the generator so do you have to be in the root directory and run the following command:
+
+```shell
+bundle exec ./bin/generate -e <exercise_slug>
+```
+
+Where `<exercise_slug>` is the same name as the slug name which is located in the `config.json` file.
+
+For more commands so can you run the following command:
+
+```shell
+bundle exec ./bin/generate --help
+```
+
+### Errors and warnings
+
+The generator will give you errors and warnings if something is wrong.
+That includes if the exercise is not in the `config.json` file, if the exercise is not in the toml file, or if the template file is missing.
+Also if it has a problem getting the `canonical-data.json` file so will it give you an error.
+The generator also uses a formatter which will give you errors if the generated file is not formatted correctly.
+The file will still be generated even if formatter gives errors, therefore can you check the file and see what is wrong and fix it in the template.
```

Comment:
```suggestion
The file will still be generated even if the formatter reports errors, So that you can check the file and see what is wrong and fix it in the template.
```